### PR TITLE
DOC: Fix type definitions in mtrand

### PR DESCRIPTION
**Pull request by bashtage**
Improve description of dtype requirements to correctly indicate float is
supported

### CODE REVIEW

File: numpy/random/mtrand/mtrand.pyx [hunk 1]
```diff
@@ -2289,7 +2289,7 @@ cdef class RandomState:
 
         Parameters
         ----------
-        df : int or array_like of ints
+        df : float or array_like of floats
             Degrees of freedom, should be > 0 as of NumPy 1.10.0,
```
Comment (bashtage) [inline numpy/random/mtrand/mtrand.pyx]: Should current docs describe old versions of NumPy?

File: numpy/random/mtrand/mtrand.pyx [hunk 2]
```diff
@@ -2210,7 +2210,7 @@ cdef class RandomState:
 
         Parameters
         ----------
-        df : int or array_like of ints
+        df : float or array_like of floats
              Number of degrees of freedom.
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: Should add the range > 0 here.

File: numpy/random/mtrand/mtrand.pyx (hunk 1 above)
Comment (charris) [reply numpy/random/mtrand/mtrand.pyx]: Yes. Can use the `.. versionchanged::` directive to do this more formally. There should be some examples in this file.

Comment (charris) [reply numpy/random/mtrand/mtrand.pyx]: But only if it changed in the code, changes in documentation don't need a note.

Comment (charris) [reply numpy/random/mtrand/mtrand.pyx]: Projects that support old versions of NumPy, SciPy for instance, need to be aware of when such changes occurred.

**Comment by charris** [comment]
Note that you can avoid the tests for document changes by adding `[ci skip]` to the commit messages, but needs to be in all of them.

**Comment by bashtage** [comment]
I wasn't sure if NumPy requires CI for docbuild

File: numpy/random/mtrand/mtrand.pyx [hunk 3]
```diff
@@ -2008,9 +2008,9 @@ cdef class RandomState:
         Parameters
         ----------
         dfnum : int or array_like of ints
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: These degrees of freedom can be `float` also.

**Comment by charris** [comment]
Small nit. Also some trailing whitespace seems to have slipped in somewhere along the line, so it would be good to clean that up too.

**Comment by charris** [comment]
Thanks @bashtage .

**Pull request merged**